recommender/engine/rl_agent/action_selector.py

```diff
@@ -28,16 +29,13 @@ def __init__(self, K: int, service_embedder: Optional[nn.Module] = None) -> None
         """
         self.K = K
         self.itemspace = self._create_itemspace(
-            load_last_module(SERVICES_AUTOENCODER).encoder
-            if service_embedder is None
-            else service_embedder
+            service_embedder or load_last_module(SERVICES_AUTOENCODER).encoder
         )
         self.forbidden_indices = self._get_service_indices(
             retrieve_forbidden_service_ids()
         )
-        assert (
-            self.I - len(self.forbidden_indices) >= self.K
-        ), "there aren't enough allowed services to recommend"
+        if self.I - len(self.forbidden_indices) >= self.K:
+            raise TooSmallRecommendationSpace
```

Comment:
I have changed this to InsufficientRecommendationSpace (as you suggested) in PR #82 - so it should be refactored during merging and later refactoring